x/go/binary/codec.go

```diff
@@ -247,3 +247,73 @@ func UnmarshalStringUint64(b []byte) (n uint64, err error) {
 	n, err = strconv.ParseUint(str, 10, 64)
 	return n, err
 }
+
+// FallbackCodec wraps a set of Codecs. When the first Codec in the chain fails to
+// encode/decode a value, it falls back to the next Codec in the chain.
+type FallbackCodec struct {
+	// Codecs is the list of codecs to fallback on.
+	Codecs []Codec
+	// FallbackOnEncode indicates whether to fallback on encoding.
+	FallbackOnEncode bool
```

Comment:
can you think of a specific situation we would want to set this to true? For forward compatibility or just if the encoding fails?

Reply:
No, I can't